blueflood-core/src/main/java/com/rackspacecloud/blueflood/inputs/formats/JSONMetricsContainer.java

```diff
@@ -142,6 +147,13 @@ public void setTtlInSeconds(int ttlInSeconds) {
             this.ttlInSeconds = ttlInSeconds;
         }
 
+        public boolean isValid() {
+            long current = System.currentTimeMillis();
+
+            // valid if collectionTime is greater than past window or less than future window
+            return ((collectionTime >= current - pastDiff ) && ( collectionTime <= current + futureDiff ));
```

Comment:
Made currentTime a param that has to be passed in for validation and initialize it for the same metric for isValid() and getValidationErrors().